src/util/file.c

```diff
@@ -298,12 +301,33 @@ int s_mkpath(char *dir, mode_t mode) {
 
     if ( is_dir(dirname(strdupa(dir))) < 0 ) {
         singularity_message(DEBUG, "Creating parent directory: %s\n", dirname(strdupa(dir)));
-        if ( s_mkpath(dirname(strdupa(dir)), mode) < 0 ) {
+        if ( s_mkpath(dirname(strdupa(dir)), mode, base) < 0 ) {
```

Comment:
A note of caution, strdupa() allocates on the stack and because this is an unbounded algo (recursive), the stack could be overflowed.